ui/perfherder/compare/CompareSelectorView.jsx

```diff
@@ -23,28 +33,54 @@ export default class CompareSelectorView extends React.Component {
     this.queryParams = this.props.$stateParams;
     this.state = {
       projects: [],
-      failureStatus: null,
+      projectsFailureStatus: null,
       originalProject: this.queryParams.originalProject || 'mozilla-central',
       newProject: this.queryParams.newProject || 'try',
       originalRevision: this.queryParams.originalRevision || '',
       newRevision: this.queryParams.newRevision || '',
       errorMessages: [],
       disableButton: true,
       missingRevision: false,
+      framework: 1,
+      frameworkName: 'talos',
+      frameworks: [],
+      frameworksFailureStatus: null,
     };
   }
 
   async componentDidMount() {
-    const { data, failureStatus } = await getData(getApiUrl(repoEndpoint));
-    this.setState({ projects: data, failureStatus });
+    const [projects, frameworks] = await Promise.all([
+      getData(getApiUrl(repoEndpoint)),
+      getData(getApiUrl(endpoints.frameworks)),
+    ]);
+    this.setState({
+      projects: projects.data,
```

Comment:
I've found a clean way to do this is to store all failure messages in `errorMessages` since it's typically shown at the top level of the app, and then you can use the `processResponse` helper in `helpers/http` to setState on the response (which will either be an updated `errorMessages` array or the actual data). Then you don't need to store the failure status.

Here's how I've used it: https://github.com/mozilla/treeherder/blob/master/ui/perfherder/graphs/TestDataModal.jsx#L63

Reply:
Done! Thanks for sharing this error handling idiom!